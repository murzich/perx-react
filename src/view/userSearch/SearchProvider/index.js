import SearchProvider from './SearchProvider';

export default SearchProvider;
