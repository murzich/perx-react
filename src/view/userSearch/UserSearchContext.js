import { createContext } from 'react';

const UserSearchContext = createContext();

export default UserSearchContext;
