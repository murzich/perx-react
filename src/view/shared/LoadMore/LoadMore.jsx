import PropTypes from 'prop-types';
import React, { useCallback, useContext } from 'react';
import { Button } from 'semantic-ui-react';
import { UserSearchContext } from '../../userSearch';

const LoadMore = ({ nextPage, fetchNextPage }) => {
  const { userName } = useContext(UserSearchContext);
  const onClick = useCallback(
    () => {
      if (nextPage) {
        fetchNextPage(userName, nextPage);
      }
    },
    [nextPage, fetchNextPage, userName],
  );

  if (!nextPage) return null;

  return (
    <Button onClick={onClick} fluid>
      load more
    </Button>
  );
};

LoadMore.propTypes = {
  nextPage: PropTypes.number,
  fetchNextPage: PropTypes.func.isRequired,
};

export default LoadMore;
