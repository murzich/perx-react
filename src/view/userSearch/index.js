import SearchForm from './SearchForm';
import SearchProvider from './SearchProvider';
import UserSearchContext from './UserSearchContext';

export { UserSearchContext, SearchProvider };

export default SearchForm;
