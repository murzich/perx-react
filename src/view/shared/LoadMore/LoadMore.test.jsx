import { cleanup, render, fireEvent } from '@testing-library/react';
import React from 'react';
import { UserSearchContext } from '../../userSearch';
import LoadMore from './LoadMore';

describe('LoadMore component', () => {
  afterEach(cleanup);

  it('should NOT render button', () => {
    const { queryByText } = render(
      <UserSearchContext.Provider value={{}}>
        <LoadMore />
      </UserSearchContext.Provider>,
    );

    expect(queryByText('load more')).toBeNull();
  });

  it('should render button is there is page value', () => {
    const { getByText } = render(
      <UserSearchContext.Provider value={{}}>
        <LoadMore nextPage={1} />
      </UserSearchContext.Provider>,
    );

    expect(getByText('load more')).toBeDefined();
  });

  it('should call fetchNextPage with userName & nextPage value on button click', () => {
    const mockOnClick = jest.fn();
    const testPage = 42;
    const testUserName = 'test-user';

    const { getByText } = render(
      <UserSearchContext.Provider value={{ userName: testUserName }}>
        <LoadMore nextPage={testPage} fetchNextPage={mockOnClick} />
      </UserSearchContext.Provider>,
    );

    fireEvent.click(getByText('load more'));

    expect(mockOnClick).toHaveBeenCalledTimes(1);
    expect(mockOnClick).toHaveBeenCalledWith(testUserName, testPage);
  });

  it('should be disabled when loading', () => {
    const { getByText } = render(
      <UserSearchContext.Provider value={{}}>
        <LoadMore nextPage={42} loading />
      </UserSearchContext.Provider>,
    );

    expect(getByText('load more')).toHaveAttribute('disabled');
  });

  it('should match snapshot', () => {
    const fakeOnClick = () => {};
    const testPage = 42;
    const testUserName = 'test-user';

    const { asFragment } = render(
      <UserSearchContext.Provider value={{ userName: testUserName }}>
        <LoadMore nextPage={testPage} fetchNextPage={fakeOnClick} />
      </UserSearchContext.Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
