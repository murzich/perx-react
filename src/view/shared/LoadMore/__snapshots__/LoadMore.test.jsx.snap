// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadMore component should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="ui fluid button"
  >
    load more
  </button>
</DocumentFragment>
`;
